packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -56,128 +74,321 @@ library LibProving {
+                tran.contester = msg.sender;
+                tran.timestamp = uint64(block.timestamp);
+
+                emit TransitionContested({
+                    blockId: blk.blockId,
+                    parentHash: evidence.parentHash,
+                    blockHash: tran.blockHash,
+                    signalRoot: tran.signalRoot,
+                    contester: msg.sender,
+                    contestBond: tier.contestBond,
+                    tier: evidence.tier
+                });
+            }
         } else {
-            return keccak256(
-                abi.encode(
-                    evidence.metaHash,
-                    evidence.parentHash,
-                    evidence.blockHash,
-                    evidence.signalRoot,
-                    evidence.graffiti,
-                    evidence.prover
-                )
-            );
+            // The new tier is higher than the previous tier, we  are in the
+            // proving mode. This works even if this transition's contester is
+            // address zero.
+
+            // zero values are not allowed
+            if (evidence.blockHash == 0 || evidence.signalRoot == 0) {
+                revert L1_INVALID_EVIDENCE();
+            }
+
+            // The ability to prove a transition is granted under the following
+            // two circumstances:
+            //
+            // 1. When the transition has been contested, indicated by
+            // tran.contester not being address zero.
+            //
+            // 2. When the transition's blockHash and/or signalRoot differs. In
+            // this case, the new prover essentially contests the previous proof
+            // but immediately validates it, obviating the requirement to set a
+            // contester, burn the contest bond, and other associated actions.
+            // This streamlined process is applied to 0-to-non-zero transition
+            // updates.
+            if (
+                tran.contester == address(0)
+                    && tran.blockHash == evidence.blockHash
+                    && tran.signalRoot == evidence.signalRoot
+            ) {
+                // Alternatively, it can be understood that a transition cannot
+                // be re-approved by higher-tier proofs without undergoing
+                // contestation.
+                revert L1_ALREADY_PROVED();
+            }
+
+            if (tid == 1 && tran.prover == blk.assignedProver) {
+                // For the first transition, if the previous prover is still the
+                // assigned prover, we exclusively grant permission to the
+                // assigned approver to re-prove the block, unless the proof
+                // window has elapsed.
+                if (
+                    block.timestamp <= tran.timestamp + tier.provingWindow
+                        && msg.sender != blk.assignedProver
+                ) revert L1_NOT_ASSIGNED_PROVER();
```

Comment:
It will revert, as within the proving window, Bob is the only one who can prove the first block, so `msg.sender` being Cecile will revert.

Reply:
Logical problem with this: Cecile's transition (well after proofWindow) will initiate the first transition.
It will set `tran.timestamp` to `block.timestamp` at line 117.

So this line will always revert in such case, because both evaluates to `true` within the if:
```
if (
                    block.timestamp <= tran.timestamp + tier.provingWindow
                        && msg.sender != blk.assignedProver
                ) revert L1_NOT_ASSIGNED_PROVER();
```

How about this:
```
if (
                    block.timestamp <= tran.timestamp + tier.provingWindow &&  block.timestamp != tran.timestamp 
                        && msg.sender != blk.assignedProver
                ) revert L1_NOT_ASSIGNED_PROVER();
```

?

Reply:
As agreed on discord, fixed with this [commit](https://github.com/taikoxyz/taiko-mono/pull/14705/commits/72b2de6136ae8b8ed466408f787d9761afc45599).